From 92692dbe956f3f5733f2842093f9d374f23b36cd Mon Sep 17 00:00:00 2001
From: "Justin M. Keyes" <justinkz@gmail.com>
Date: Sat, 12 Jul 2025 11:41:55 -0400
Subject: [PATCH] fix(health): bad format() call

Problem:
Bad format() call on PUC Lua #34901

    Error: Failed to run healthcheck for "vim.health" plugin. Exception:
    runtime/lua/vim/health/health.lua:89: bad argument #1 to 'format' (string expected, got nil)

Solution:
Avoid passing nil.
---
 runtime/lua/vim/health/health.lua | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/runtime/lua/vim/health/health.lua b/runtime/lua/vim/health/health.lua
index 6cb772118ae1f0..e942ee789c93bf 100644
--- a/runtime/lua/vim/health/health.lua
+++ b/runtime/lua/vim/health/health.lua
@@ -87,9 +87,9 @@ local function check_config()
     health.error(
       'Locale does not support UTF-8. Unicode characters may not display correctly.'
         .. ('\n$LANG=%s $LC_ALL=%s $LC_CTYPE=%s'):format(
-          vim.env.LANG,
-          vim.env.LC_ALL,
-          vim.env.LC_CTYPE
+          vim.env.LANG or '',
+          vim.env.LC_ALL or '',
+          vim.env.LC_CTYPE or ''
         ),
       {
         'If using tmux, try the -u option.',
